### PR TITLE
Refactor Line Merging of Equations, on Backspace. 

### PR DESCRIPTION
**Pull request by matthewborden**
When there is an equation at the end of a line and we are backspacing
from the line below. Instead of trying to treat the equation at the end
of a line as a special case (and delete it), we place the cursor at the end of the
equation and then let it be handled as the case of deleting a equation
within a line, thus the user presses backspace once to go up a line and 
then again to delete an equation.

### CODE REVIEW

File: build/squire-raw.js [hunk 1]
```diff
@@ -3364,7 +3342,28 @@ var keyHandlers = {
...
                 }
-                self.setSelection( range );
+                // Get the common ancestor of the range.
+                var ancestor = range.commonAncestorContainer;
+                
+                /*  Check if the common element to the range passed contained a mathjax block? If it didn't and was a svg 
+                    element we treat it as an element that is a descendant of a <svg>. We loop up the tree until we find an 
+                    element with the mathjax class and then use that to set the caret after the equation.
+                */
+                if (ancestor.querySelector && !ancestor.querySelector("mathjax") && checkInSVG.test(ancestor)) {
```
Comment (matthewborden) [inline build/squire-raw.js]: Should test ancestor.nodeName

**Comment by djrodgerspryor** [comment]
The changes look good, but can we not commit the build files?

Just add build/ to .gitignore

**Comment by matthewborden** [comment]
Just to clarify, build/squire.js is the file used in production.

We could use https://www.npmjs.com/package/gulp-exec to run `make` during the web client build process, that would generate the build directory. Another option would be a githook that reminds us to run `make` before committing and then keep the build directory in the repo. 

Note: The Squire project currently commits the build directory. https://github.com/neilj/Squire

**Comment by matthewborden** [comment]
Closing until changes to the build process are merged.

**Pull request merged**